### PR TITLE
nixVersions.nix_2_11: init at 2.11.0

### PR DESCRIPTION
**Pull request by zowoq**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

`nix-fallback-paths` done in https://github.com/NixOS/nixpkgs/pull/188772

### CODE REVIEW

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/tools/package-management/nix/default.nix [hunk 1]
```diff
@@ -84,7 +84,13 @@ in lib.makeExtensible (self: {
     patches = [ ./patches/flaky-tests.patch ];
   };
 
-  stable = self.nix_2_10;
+  nix_2_11 = common {
+    version = "2.11.0";
+    sha256 = "sha256-9+rpYzI+SmxJn+EbYxjGv68Ucp22bdFUSy/4LkHkkDQ=";
+    patches = [ ./patches/flaky-tests.patch ];
```
Comment (lovesegfault) [inline pkgs/tools/package-management/nix/default.nix:90]: Is this still needed?

Comment (zowoq) [reply pkgs/tools/package-management/nix/default.nix:90]: yes

**Review by lovesegfault — CHANGES_REQUESTED** [review]
You need to update the fallback paths too.

LGTM otherwise.

**Comment by zowoq** [comment]
> You need to update the fallback paths too.

If you read the top post you'll see that this in another PR.

**Comment by lovesegfault** [comment]
> > You need to update the fallback paths too.
> 
> 
> 
> If you read the top post you'll see that this in another PR.

I'm on my phone, so it was hidden.

Just cherry-pick into this branch so they land together.

**Comment by bjornfor** [comment]
> nixVersions.stable: 2.10 -> 2.11

Observation: If that commit was a separate PR it'd make backporting to stable easier (just label the PR).

**Comment by SuperSandro2000** [comment]
> Observation: If that commit was a separate PR it'd make backporting to stable easier (just label the PR).

Then we would need to merge 3 PRs at the same time. Not very practical.

**Comment by bjornfor** [comment]
> Then we would need to merge 3 PRs at the same time. Not very practical.

AFAICT it's 2 PRs, and they dont have to be merged at the same time, but we must ensure they are merged in correct order.

**Comment by zowoq** [comment]
> Observation: If that commit was a separate PR it'd make backporting to stable easier (just label the PR).

- two PRs to master, automated backport (must be merged in order)
- one PR to master, manual backport

So either way people still need to do two PRs except opening two on master is more complicated?

**Comment by SuperSandro2000** [comment]
> but we must ensure they are merged in correct order.

I don't think that is worth it for the little convince improvement.

**Pull request merged**